### PR TITLE
Add dev-preview multi-arch 4.13 to nightly 4.13 upgrade

### PR DESCRIPTION
**Pull request by shellyyang1989**
Refers [MIXEDARCH-133](https://issues.redhat.com/browse/MIXEDARCH-133)

### CODE REVIEW

**Comment by shellyyang1989** [comment]
INFO[2023-02-07T06:04:25Z] Step aws-ipi-ovn-ipsec-p2-f14-cucushift-upgrade-toimage succeeded after 1h23m30s. 

We need to migrate dev-preview to stable builds once stable is available.

cc @aleskandro @jianlinliu @liangxia @jhou1 @YuLi517 for review. Thanks!

**Comment by aleskandro** [comment]
@shellyyang1989 is this going to happen for amd64 and arm64 jobs as well?

File: ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.13__multi-nightly-4.13-upgrade-from-stable-4.13.yaml [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+    tag: latest
+build_root:
+  image_stream_tag:
+    name: release
+    namespace: openshift
+    tag: golang-1.18
+releases:
+  latest:
+    prerelease:
+      architecture: amd64
```
Comment (aleskandro) [inline ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.13__multi-nightly-4.13-upgrade-from-stable-4.13.yaml:46]: should this be multi?

File: ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.13__multi-nightly-4.13-upgrade-from-stable-4.13.yaml [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+      product: ocp
+      stream: 4-dev-preview
+      version_bounds:
+        lower: 4.13.0-0
+        upper: 4.14.0-0
+  multi-latest:
```
Comment (aleskandro) [inline ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.13__multi-nightly-4.13-upgrade-from-stable-4.13.yaml:52]: you'd test by removing this stanza and using multi in the architecture field above.

File: ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.13__multi-nightly-4.13-upgrade-from-stable-4.13.yaml [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+        and @network-ovnkubernetes and not @fips'
+    test:
+    - ref: ipi-install-heterogeneous
+    - chain: openshift-upgrade-qe-test
+    workflow: cucushift-installer-rehearse-aws-ipi-ovn-ipsec
+zz_generated_metadata:
+  branch: release-4.13
+  org: openshift
+  repo: openshift-tests-private
+  variant: multi-nightly-4.13-upgrade-from-stable-4.13
```
Comment (liangxia) [inline ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.13__multi-nightly-4.13-upgrade-from-stable-4.13.yaml:108]: The name is a little confusing, we are not testing stable 4.13 at the moment as we do not have stable 4.13 yet, the variant should be `multi-nightly-4.13-upgrade-from-dev-preview-4.13` or `multi-nightly-4.13-upgrade-from-ec-4.13`

File: ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.13__multi-nightly-4.13-upgrade-from-stable-4.13.yaml (hunk 2 above)
Comment (shellyyang1989) [reply ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.13__multi-nightly-4.13-upgrade-from-stable-4.13.yaml:52]: Per what you suggested in last meeting, you'd prefer to minimum the files. So I was creating the file to hold all the upgrade to multi-arch 4.13 from both single-arch and multi-arch stable 4.13. If you'd prefer to add a separate file to hold single-arch 4.13 to multi-arch 4.13, it's fine with me updating based on your comments.

Comment (aleskandro) [reply ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.13__multi-nightly-4.13-upgrade-from-stable-4.13.yaml:52]: Hi @shellyyang1989, my suggestion is to define single-arch to multi-arch upgrade/migration testing in the related single-arch test config. For example, arm64-to-multi can be defined in the arm64 upgrade test config files.

Comment (shellyyang1989) [reply ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.13__multi-nightly-4.13-upgrade-from-stable-4.13.yaml:52]: It's fine with me!

**Comment by shellyyang1989** [comment]
Refers [MIXEDARCH-166](https://issues.redhat.com/browse/MIXEDARCH-166)

**Comment by aleskandro** [comment]
Thanks @shellyyang1989!

/lgtm

File: ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.13__multi-nightly-4.13-upgrade-from-multi-ec-4.13.yaml [hunk 4]
```diff
@@ -0,0 +1,111 @@
+base_images:
```
Comment (liangxia) [inline ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.13__multi-nightly-4.13-upgrade-from-multi-ec-4.13.yaml:1]: We do not support cross arch upgrade, so no need to stat `multi` twice, to shorten the job names (as file name is part of the job name), also to keep the same name style as the other arch.

**Comment by shellyyang1989** [comment]
@liangxia Could you please help merge it? Thanks!

**Pull request merged**